the-collab-lab/tcl-21-smart-shopping-list
# th vs firebase config

## Description

1. Added firebase and react-firestore-hooks dependencies.
2. Enabled exporting of Firestore from src/lib/firebase.js to allow for easy importing in other components.
3. Added a components directory for organization.
4. Added a component to test Firestore read, write, and delete functionality.

## Related Issue

closes 1

## Acceptance Criteria

* The following have been added as project dependencies: firebase & react-firebase-hooks
* Make a change in the Firestore database and it shows up in the app
* Make a change in the app and it shows up in the Firestore database

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓  | :link: Update dependencies |
|    | :scroll: Docs              |

## Notes
While we know it is good practice to create separate files for separate React components, since the components we added are tests, we decided to house them all in a single throwaway file: src/components/FirestoreTest.js.

## Testing Steps / QA Criteria
1. Pull down remote branches: `git pull`
2. Navigate to correct branch: `git checkout th-vs-firebase-config`
3. Run server and view in browser on localhost:3000: `npm start`
4. In the browser, use the form to add a new shopping list and check to see the new list is displayed above the form.
5. Click the button next to an item and see that item is deleted.
6. In a separate browser window, log into Firebase Console, go to our project, and view our Firestore Database.
7. Check that changes made via the app are reflected in Firestore.
8. Try adding a new document to the 'lists' collection in Firestore (you will need to enter a 'name' (string) field and a 'createdAt' (timestamp) field, for the website orders items by the time they were createdAt and displays them by their name.); check to see if you new list populates in app.
